solv/simplequery.c

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2022 VMware, Inc. All Rights Reserved.
+ *
+ * Licensed under the GNU Lesser General Public License v2.1 (the "License");
+ * you may not use this file except in compliance with the License. The terms
+ * of the License are located in the COPYING file of this distribution.
+ */
+
+#include "includes.h"
+
+#define ASSERT_ARG(x) { \
+    if (!(x)) { \
+        dwError = ERROR_TDNF_INVALID_PARAMETER; \
+        BAIL_ON_TDNF_LIBSOLV_ERROR(dwError); \
+    } \
+}
+
+#define ASSERT_MEM(x) { \
+    if (!(x)) { \
+        dwError = ERROR_TDNF_OUT_OF_MEMORY; \
+        BAIL_ON_TDNF_LIBSOLV_ERROR(dwError); \
+    } \
+}
+
+/* Split nevra into name, ever, arch. This modifies its input,
+   and returns pointers to the respective components */
+static
+int split_nevra(char *nevra, char **name, char **evr, char **arch)
+{
+    char *p = nevra + strlen(nevra) - 1;
```

Comment:
validate function args

Reply:
This is a static helper function. I don't think we need to check function args for every function.